gateway/cert.go

```diff
@@ -274,8 +316,10 @@ var tlsConfigMu sync.Mutex
 
 func getClientValidator(helloInfo *tls.ClientHelloInfo, certPool *x509.CertPool) func([][]byte, [][]*x509.Certificate) error {
 	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+		// don't return error if not cert provided, as request could be for URL which does
+		// not require a cert
 		if len(rawCerts) == 0 {
-			return errors.New("x509: missing client certificate")
```

Comment:
`getClientValidator` function runs only mode set to `tls.RequireAndVerifyClientCert`. E.g it should do the full mTLS flow.

Reply:
For the API I attached it listens on `/dom/` with `spec.Domain = ""` if a request is made to `https://localhost/not-listen-path` `tls.RequireAndVerifyClientCert` is still set and expects a certificate, that's why I made it acceptable for nil cert.

Reply:
Yes, but this API has `use_mutual_tls_auth` set to true. So mTLS is requirment in this case, and users have to pass certificate otherwise they get TLS error. 

If, for example, you had second API which also `spec.Domain = ""`, and different listent path, but it does not require mTLS, and mode will be set to `tls.RequestClientCert` and `getClientValidator` will not be executed at all. Because Gateway can't decide if it require if it is true mTLS, or it has to fallback to validating certificate on HTTP middleware level.

Reply:
Ah ok, this makes sense as I was only testing with one API loaded. I've reverted the change.